### PR TITLE
[3.4] client/v3/watch.go: use fmt go pkg for gRPC metadata map printing

### PR DESCRIPTION
**Pull request by mohamedawnallah**
## Description

This PR backports the changes from #18308 to the release-3.4 branch.

## Issue

Addresses Issue #18303 in the backport release-3.4.

### CODE REVIEW

**Review by ahrtr — APPROVED** [review]
LGTM

Thanks

**Comment by ivanvc** [comment]
/retitle [3.4] client/v3/watch.go: use fmt go pkg for gRPC metadata map printing

/ok-to-test

**Comment by jmhbnz** [comment]
Closed and re-opened to trigger tests. GitHub did not run them initially.

File: clientv3/watch_test.go [hunk 1]
```diff
@@ -15,9 +15,12 @@
 package clientv3
 
 import (
+	"context"
 	"testing"
 
-	"go.etcd.io/etcd/mvcc/mvccpb"
+	"google.golang.org/grpc/metadata"
+
+	"go.etcd.io/etcd/api/v3/mvccpb"
```
Comment (jmhbnz) [inline clientv3/watch_test.go:23]:
```suggestion
	"google.golang.org/grpc/metadata"

	"go.etcd.io/etcd/mvcc/mvccpb"
```

**Review by jmhbnz — CHANGES_REQUESTED** [review]
Thanks for raising this backport @mohamedawnallah. Our `release-3.4` codebase is structured quite differently from `main` or `release-3.5` in terms of modules so you will need to adjust as follows.

File: clientv3/watch_test.go (hunk 1 above)
Comment (mohamedawnallah) [reply clientv3/watch_test.go]: Great catch! updated the relevant commit.

**Review by jmhbnz — APPROVED** [review]
LGTM - THanks @mohamedawnallah

**Review by ivanvc — APPROVED** [review]
LGTM. Thanks, @mohamedawnallah.

**Pull request merged**